Common/include/toolboxes/graph_toolbox.hpp

```diff
@@ -495,3 +518,77 @@ T colorSparsePattern(const T& pattern, size_t groupSize = 1,
   /*--- Move compressed coloring into result pattern instance. ---*/
   return T(std::move(colorPtr), std::move(outerIdx));
 }
+
+
+/*!
+ * \brief Create the natural coloring (equivalent to the normal sequential loop
+ *        order) for a given number of inner indexes.
+ * \note This is to reduce overhead in "OpenMP-ready" code when only 1 thread is used.
+ * \param[in] numInnerIndexes - Number of indexes that are to be colored.
+ * \return Natural (sequential) coloring of the inner indices.
+ */
+template<class T = CCompressedSparsePatternUL,
+         class Index_t = typename T::IndexType>
+T createNaturalColoring(Index_t numInnerIndexes)
+{
+  /*--- One color. ---*/
+  su2vector<Index_t> outerPtr(2);
+  outerPtr(0) = 0;
+  outerPtr(1) = numInnerIndexes;
+
+  /*--- Containing all indexes in ascending order. ---*/
+  su2vector<Index_t> innerIdx(numInnerIndexes);
+  std::iota(innerIdx.data(), innerIdx.data()+numInnerIndexes, 0);
+
+  return T(std::move(outerPtr), std::move(innerIdx));
+}
+
+
+/*!
+ * \brief A way to represent one grid color that allows range-for syntax.
+ */
+template<typename T = unsigned long>
+struct GridColor
+{
+  static_assert(std::is_integral<T>::value,"");
+
+  const T size;
+  const T* const indices;
+
+  GridColor(const T* idx = nullptr, T sz = 0) : size(sz), indices(idx) { }
+
+  inline const T* begin() const {return indices;}
+  inline const T* end() const {return indices+size;}
+};
+
+
+/*!
+ * \brief A way to represent natural coloring {0,1,2,...,size-1} with zero
+ * overhead (behaves like looping with an integer index, after optimization...).
+ */
+template<typename T = unsigned long>
+struct DummyGridColor
+{
+  static_assert(std::is_integral<T>::value,"");
+
+  T size;
+  struct {
+    inline T operator[] (T i) const {return i;}
+  }
+  indices;
+
```

Comment:
When compiling without OpenMP support the vector of colors is a static array with one color of this dummy type, which after optimization is perfectly equivalent to a plain for loop with an integer variable (see operator [] above).